libsubid/subid.h.in

```diff
@@ -55,6 +55,19 @@ extern "C" {
  */
 bool subid_init(const char *progname, FILE *logfd);
 
+/*
+ * subid_free: free memory allocated in any subid_* function
+ *
+ * @ptr: Pointer to a memory block to release.
+ *
+ * Some functions like @subid_get_uid_ranges allocate memory internally. As
+ * soon as a result is no longer needed, it should be freed with this routine.
+ * Initially, default function `free()` was used. Thus for backward
+ * compatibility this function falls back to `free()` if a plugin does not
+ * explicitly specify routine to free allocated memory.
+ */
+void subid_free(void *ptr);
+
```

Comment:
Now that we have a deallocator, should we use `[[gnu::malloc(deallocator)]]` for documenting which functions should be freed by this one?